From 23b42785f9e8d83f55d55cb56bc66feddf912cc0 Mon Sep 17 00:00:00 2001
From: Pavithra R <pavir@codeaurora.org>
Date: Thu, 9 Aug 2018 21:48:44 +0530
Subject: [PATCH] [testmgr] Update test vector for md5

NSS crypto does not support authentication key size greater than hash
algorithm block size in case of HMAC mode. So, the test vectors with
authentication key size greater than block size are under an if check.

Change-Id: I93087d1589bcab8708512b0dc57112b43918a921
Signed-off-by: Pavithra R <pavir@codeaurora.org>
Signed-off-by: Tushar Ganatra <quic_tganatra@quicinc.com>
---
 crypto/testmgr.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/crypto/testmgr.h b/crypto/testmgr.h
index 025dfcb02ece..c980cf4369c0 100644
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -7132,7 +7132,10 @@ static const struct hash_testvec hmac_md5_tv_template[] =
 		.psize	= 20,
 		.digest	= "\x56\x46\x1e\xf2\x34\x2e\xdc\x00"
 			  "\xf9\xba\xb9\x95\x69\x0e\xfd\x4c",
-	}, {
+	},
+#ifndef CONFIG_CRYPTO_DISABLE_AHASH_LARGE_KEY_TEST
+	/* TODO : Require support for Auth key size larger than block length */
+	{
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -7160,6 +7163,7 @@ static const struct hash_testvec hmac_md5_tv_template[] =
 		.digest	= "\x6f\x63\x0f\xad\x67\xcd\xa0\xee"
 			  "\x1f\xb1\xf5\x62\xdb\x3a\xa5\x3e",
 	},
+#endif
 };
 
 /*
-- 
2.34.1

